src/algorithms/calorimetry/CalorimeterHitDigi.cc

```diff
@@ -181,10 +182,14 @@ void CalorimeterHitDigi::process(
                      std::pow(m_cfg.eRes[2] / (edep), 2)
                   )
                 : 0;
-        double    corrMeanScale_value = corrMeanScale(leading_hit);
-        double    ped     = m_cfg.pedMeanADC + m_gaussian(m_generator) * m_cfg.pedSigmaADC;
-        unsigned long long adc     = std::llround(ped + edep * corrMeanScale_value * ( 1.0 + eResRel) / m_cfg.dyRangeADC * m_cfg.capADC);
-        unsigned long long tdc     = std::llround((time + m_gaussian(m_generator) * tRes) * stepTDC);
+        double corrMeanScale_value = corrMeanScale(leading_hit);
+
+        // Pedestal in ADC channels must be positive
+        double ped = std::max(m_cfg.pedMeanADC + m_gaussian(m_generator) * m_cfg.pedSigmaADC, 0.0);
+
+        // Note: both adc and tdc values must be positive numbers to avoid integer wraparound
+        unsigned long long adc = std::llround(ped + edep * corrMeanScale_value * (1.0 + eResRel) / m_cfg.dyRangeADC * m_cfg.capADC);
```

Comment:
I'm not aware why noise contribution has to be positive in our modeling.
```suggestion
        double ped = m_cfg.pedMeanADC + m_gaussian(m_generator) * m_cfg.pedSigmaADC;

        // Note: both adc and tdc values must be positive numbers to avoid integer wraparound
        unsigned long long adc = std::max(std::llround(ped + edep * corrMeanScale_value * (1.0 + eResRel) / m_cfg.dyRangeADC * m_cfg.capADC), 0);
```